### PR TITLE
[W01] Add test case executor

### PR DESCRIPTION
**Pull request by Anatoly03**
#### General

- [ ] I locally pass **all** tests.
- [x] I calculated manually the correct solutions on paper.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://github.com/JohannesStoehr/gad23-tests/blob/main/CONTRIBUTING.md).
- [x] The newly added tests do not reveal the solutions dynamically calculating the solutions as expected in the exercise.

### Description

- add `cases.txt`: Text Format of Test Case Data
- add `TestExecutor.java` Add Test Case Collector & Test Case Executor

#### Review
- [ ] Review 1
- [ ] Optional Review 2

### CODE REVIEW

**Comment by Anatoly03** [comment]
Note with this PR:  I locally **DO NOT** pass all tests due to an IDE error, **BUT** calculated manually the correct solutions on paper. I'd still prefer not to check that box before other students will be able to tell me for safety.

**Comment by Anatoly03** [comment]
The current test cases have been added:

```
200 - 10 - 9
▌▀▜▀▀▀▀▛▀▛▜▛▀▀▜▀▛▛▀█▛▛▜▜▀▜▀▀▜▛▀▀▀▀▛▀▀▛▛▀▀▀▜▀▀▀▀▀▀▀▜▀▛█▛▜▀█▀▀▀▜▜▜▜▀▀▀█▜▛▜▛▀▜▀▀▀▛▜▀▀▛█▀▀▀▀▀▛▀▜▀▀▀▛▀▀▀▜
▌▀▞▝▀▗▝  ▝▗▌▚▝▗▙▗  ▞▖▗  ▄▘ ▞ ▗▖▖▐▐▝    ▜▟▄▝▘  ▄ ▝ ▟▗ ▐▝▘▖▐   ▝▝ ▗▘▖▐▖▐ ▖▄▐▝  ▗▛▘ ▚▛▞▛▞▖▘ ▌▖▗▝▘▚▜▐▖ ▐
▙ ▘ ▌▝▌▗ ▄▝ ▗▟▀   ▝▌ ▚  ▐ ▙ ▘▀▐▝▘▖▚▗▘▗ ▀▘▘▗▌  ▘▖▗▘▚  ▖▖▌▘▞▖ ▘ ▞ ▝▄▖ ▖▝▟ ▞▝▝▘▘  ▖▗▖▙▘▖ ▐▟▚▖▖ ▖█▘  ▗▄▐
▙▝▐ ▗ ▚ ▖▖▌▖ ▝▝  ▘▖  ▖ ▐▖▝   ▗▌ ▖ ▙▘ ▖▄   ▗▘▖  ▘▄ ▗ ▐▞▝▗▘ ▖▜▖▘ ▐▖▖▄▞▝▚▗▝▝    ▘▘   ▗▗▌▗ ▚▚▌   ▀▖▘  ▖▐
▙▟█▟▟▟▄▙▟▄▙▟█▙█▄▄▄▙▄▄▄▄▄▟█▄▙▄▟▄▟▄▄▄▄▄▄▙▄▄▙▟▟▟▟▙▙█▟▟▙▄▟▄█▙▄▙▟▄▄▄▟▄▙▄▄▄▄▙▄▄▄▄▙▟▟▄▟▟▟▙▟█▙▄▄▄▄▙▄▄▟▟▟▄▄▄▄
```

```
200 - 10 - 6
▌▀█▛▛▜▛▀▀▜▀▀▀▀▛▀▀▛▀▛▛▀▀▀▀▀▀▀▀▜▛█▀▛▀▀▛█▀▀▀▜▀▀█▛▛▛▛▀▀▛▀▀▀▛▀▀▀▛▀▀▛▜▀▀▀▛▀▀▀▀▀▛▀▀▀▀█▀▛█▜▀▀▜▜▀▀▜▛▀▀▀▀▀▀▛▛▜
▌▝▌ ▝▘▟▄▞▗▌▄▖▟▝▜▖▗▖  ▘▙▙ ▖▘▚ ▝▗▐▀  ▖  ▝▖▝ ▘▖▄ ▖  ▘ ▗▚▘▐▙▐ ▀ ▐▖ ▖ ▗▜ ▗ ▄▝ ▐  ▖▘▘ ▘▌▖▚▝ ▖ ▚▗▐▖  ▛ ▞▘▖▐
▙▄▐▝▘     ▟▝▘▝▝▘▖ ▘▄ ▖  ▖▝▚  ▜ ▝▖ ▖▗ ▖▌▜▘ ▗▀ ▘ ▘▘ ▗▖ ▞▗ ▝  ▞▘ ▘ ▀ ▖ ▘▄▚ ▝▖▝▖▝▝▙▚▞▗ ▟▚ ▝▖▄▀▖▘ ▘▞▗▝ ▟▐
▛▗▝▚▞▐▀▄ ▗  ▖▝▘    ▚▀▞▄▗▐▗▀ ▘█ ▚▌▝▙▚▌▐ ▘▖▘ ▄  ▐   ▖ ▙▘▜ ▖  ▝ ▌▘▗█▘  ▘▚▞  ▀▘▚▗ ▀▐▘▝▝▟▗▀ ▗▖▗▛ ▛ ▌▖▞▚▘▟
▙▄▄▄▄▄▄▟▙▄▄█▄▄▙▟▄▄▄█▟▄▄▄▙▙▄▟▟▄▄▄▄▟▄▄▙▙▄▄▄▟▄▄▟▄▄▄▄▄▄▟▟▄▙▟▟▄▙▄▄▟▄▟▄▟▙▄█▄▄▟▄▙▄▄▄▄▙▄▄▄▄▟▄▄▄▄▄▟█▙▄▙▟▄▄▄▙▄
```

```
200 - 50 - 9
▌▀▀▀█▀▀▀▜▀▀▜▜▀▛▀▀▀▀▛▀▛▛▜▜▀▜▀▀▀▀▜▀▀▀▜▀▀▀▀▜▜▀▀▀▀▛▜█▀▀▀▛█▀▀▛▀▀▛▜▛█▀▛▀▀▀▀▀▀▜▀█▜▛▀▛▛▜▀▀█▀▀▀▛▀▜▀▀▛▀▀▜▜▀▀▀█
▛ ▗▌▖▀▟ ▞ ▖▜▝▜▗▗▝▖▞▘▄▗▚ ▗  ▖ ▚ ▐  ▜▟ ▝ ▞▚▖▘ ▖▘▖▜▝ ▐ ▝ ▞▖▖▜ ▌▖▝▀▖▌▐▌▐▄▖▗▖ ▘▖▘▚▞ ▌▗▝ ▝▘▀▝ ▘▛▝ ▝▞▖▖▖ ▞▜
▌▗▗    ▗▚ ▘▛▀▖▞▘▗▐▖▗▀▀▝  ▐▐▖▄ ▘▗▝ ▝▘  ▄▗▗▀▐ ▗ ▘▗ ▞▘▞▘ ▗▘ ▝▗ ▝▝ ▝▚ ▗▗▖█  ▘▗ ▚ ▐ ▌  ▘▄▖▖▝▖▝ ▖ ▀▄▗▚▝▞▖▐
█ ▌   ▌▘▗ ▐ ▖▞▝ ▗▗ ▘▌▗ ▚▖▗ ▖▝ ▗ ▘ ▐▘  ▛  █▖▞  ▚ ▗▘ ▘▖▝ ▝▝▄▌▗▄▌▝ ▚▗▐ ▘ ▘  ▗▐  ▝▛ ▚▘  ▐ ▖ ▝ ▘ ▖▚ ▗▞ ▖▐
▛▗▀  ▌▖ ▞▗ ▘ ▚▘▀ ▘▘ ▘▚▄▘▄▗▘▌▐▀▝▞ ▗▀▚▝ ▖ ▛ ▖▞▖ ▘ ▖▘ ▚ ▗  ▖▐▙▘▝▖▖▗▄▝▘▘▗ ▘▗ ▗ ▄▗▀▗▗ ▘▗▛▀ ▘  ▖ ▜ ▌▄▀▘ ▗▐
▌▘▖ ▝▖ ▜▝ ▄▖ ▗ ▗▚▘▘ ▄▀▗ ▝▖ ▄ ▄▝ ▗▝▛▞▞▖ ▝▝  ▝     ▖ ▝▌▝▝▖ ▌▝ ▘▖  ▝▝ ▞▛▐▚ ▘▄  ▖ ▘▘▐  ▗▗ ▞▛▗ ▖▗▝▘ ▗▚▝▖▐
▌▖▖▖    ▛ ▌  ▗ ▐▗▘▝▌    ▘ ▗▖ ▗▝▘▝▗ ▗ ▗▘ ▐▐▗▄ ▘▝▌▚ ▞ ▞▗   ▗▝ ▘▝▗▝▘ ▖▐ ▖▗▗▘▝▗▗▘▀▖ ▞▘▝▖▐▙ ▀  ▖▌▝▐▌  ▚▚▟
▌▘▚  ▝▙▖▐ ▄▗ ▄▌ ▞▝▐▗ ▖▗▝▝▗ ▝▗▞ ▄▝   ▀▗▚▗▄▝▚  ▘  ▚▘ ▄▄▀▘  ▘▝▖   ▘ ▝ ▖▜ ▀▀▌▌▖▝▘▚▝▝▗▖▖▚▐  ▚▖▖▙▝  ▖▝  ▘▟
▌▗ ▗ ▘ ▗▝▗▄▄  ▖▝ ▞▝▗ ▐▀ ▝▗▘ ▚▖▀▖▗▄    ▖▚▀▗▄▘▗▗▚  ▘▝▗▖▝▐▌▝▝ ▗▘▄ ▘▜▖▚ ▝▛▘▘ ▗▌▘ ▀▖  ▗▖▗ ▘█ ▞  ▀▄▖▗▜ ▙▞▜
▙▘▟▟▄▘ ▝▖▀▘▜▖ ▗ ▚ ▖▘▖▘▖▖█ ▗▘  ▝▛▝▀█▘ ▝▝▘▖ ▘▞   ▘ ▖▗ ▄▌ ▐▙▗ ▗▟▖▘   ▖▞▄▖▞ ▘▖ ▝  ▗▗▖▟▌  ▖ ▖▌▗▛▖ ▗▘ ▗▖ ▐
▌▘▌▌▄▚▝   ▚▘▚▗ ▀▐▀▀ ▗ ▜▟▝ ▖▖▖ ▌▘▄▗▟ ▘ ▞▚ ▖▀▐▝▗▝▖▘▐▝▘▝▞ ▖▖ ▚ ▘ ▖▝▚▝▗▞▘▘▖▘  ▞▄  ▗▘ ▌▘▝▗▞ ▗▚▌▘▖▖▗▖▘▀▘▄▟
▛ ▛   ▟ ▖▞▐▖ ▖▟▝▀▖   ▝▜▖▗▘ ▗▘▜ ▖  ▘▖▞  ▖▄▝▗▖ ▘  ▗▝  ▗ ▘▐ ▘▘ ▗▖▗▖▙▖▀    ▝▗▜▗▞█ ▗▄▚▝▝  ▗▗     ▘▗ ▜▙ ▝▐
▌▌▐▝▘▌ ▘▘▚ ▘ ▖▝▘▖▗▐▗ ▐▚▜▝ ▖     ▛▗▘▚▝  ▀ ▖▗▗▚▛  ▗ ▝  ▘ ▖▘   ▀▝▘▘ ▝  ▖▟▄ ▖ ▗  ▘▝▘▝▖▗▖▖ ▖▝▀   ▐▘▌▝ ▝▘▐
▌▘▞ ▟  ▗ ▗▝▘▜▞  ▗▗▝ ▚▀   ▞▌▖▞▚ ▖▘ ▖▀▘▄  ▝ ▖▖ ▘  ▘   ▄▘▗▗ ▘  ▗▗▗▗ ▝▘ ▚▞▖▝▝▗ ▗ ▗ ▘▟▐ ▖▙▀▗▚▝▘  ▚▘▗▘▛ ▝▟
█▐ ▖ ▗▖ ▛▀▀▘▝▖  ▞ ▝▖▗▜▖   ▐▘  ▝▄ ▗▝▗▗▘▗▘ ▄▀▗▗ ▘ ▌ ▝ ▛▝▝▖ ▝  ▖  ▞  ▌▀ ▘ ▝▖▗▝▘ ▚▖▐▀▝▘▗▟ ▐▗ ▀ ▖ ▝▗▝▜ ▘▜
▛▐▝▗ ▐▚▘ ▝▗ ▛▘▞▀▘▗▖▘▗█▝    ▄▞▄▄▘▗▖▗▙ ▝▚▗▄▄  ▖▘▘▖▝▝  ▖▄▞▖▘▙▘▖▝▖ ▝ ▙▞ ▗▘▘ ▖▝▘ ▀  ▘▞▝▙  ▖▜ ▘▞▖ ▖  ▘▟ ▘▟
▛▘ ▄▄  ▝▖▖▖ ▘  ▝▟▐▞▌ ▘ ▖▀▝ ▖▗ ▞▖▀▄▖▐▝▘▛▌    ▗▜▝  ▞▘▝▝ ▝▚▐▘▞▌▖▌▝▝▖  ▘ ▄▝▞▖ ▌▌  ▄▝▗ ▖▘ ▜▝  ▖▄▗ ▀ ▐▗▖ ▐
█▗▝▐▖▞▀▝ ▝▗    ▗▗▙▝  ▖ ▗▀ ▗ ▝▝ ▖▝▚▝  ▀ ▚▖ ▐▘▗▐▖  ▝▝▖▙▌▗▖▀▖  ▖▖ ▖ ▗▞▝   ▝ ▖   ▙▝▚ ▘▙ ▘▖▐▝▝▞▗▖▖▖▚▖  ▖█
▌▄  ▗ ▐ ▗  ▝▗▞▘ ▐▐▗    ▘▙▗ ▝ ▝▖▘ ▘▖▀ ▞ ▖ ▘  ▚ ▗▜ ▐   ▖▀ ▄▗▐ ▀ ▚▘ ▗ ▝▗▚ ▗ ▖▌▚▞▝  ▝▞▘  ▐▝ ▝ ▞▗▗ ▚ ▗▘▝▐
▌▖▀ ▗▗▌▄▖ ▄  ▗▖▘▀  ▗▗▗▗▀▘▖ ▚ ▘ ▘▙ ▝▗▝▀▀▟  ▖  ▖▚▖  ▐▀▚ ▝ ▝▗▘▚▝▘▘▖▖▖ ▀ ▖▝▗▐▝  ▝ ▖▗▝ ▖▐▗▌ ▝ ▀▗▝▝ ▖▗ ▘▝▟
▛▞▗▟  ▝▖ ▘▌▞▗▗▄▐▞   ▐▞▖▐▖▖▘ ▖▗▀▖ ▐▚ ▐▖▘ ▟▞▀  ▗▘ ▗▘▗▟▖▄▗▝▖▀ ▛▖ ▚▘▗ ▘ ▌▐▗▖   ▖ ▀█▗▗▝▝▘▖ ▖    ▘▚▘▝▌▖  ▐
▛▝▚▝▗▄▝▐  ▝ ▗▐▞ ▛▘▗▄ ▝     ▝▗▝▐▗▖▖▝   ▀  ▖   ▜▗ ▟▘▚▚▌▝▖  ▝ ▀ ▗▝▖▘▖▖ ▘▀▞  ▖▗▖▗ ▜▗  ▝  ▝█ ▖   ▖▙ ▗▝  ▐
▌▗▝ ▐▝▚▝ ▖ ▖▞▝▞ ▌▖▖▝▝   ▄▟▝▝▐▖ ▖▗▘▄ ▗▖ ▞▘▚▘▗▙▝▐▀▝ ▚▙▗ ▝ ▖▞▖  ▛▝ ▗  ▘▘▗  ▝▖▜▗▞  ▘ ▖▖▄▖ ▛▝▝▝▖▌▌  ▝ ▞▖▐
▙ ▝▘ ▗▄▖ ▘▝▐ ▄ ▘ ▘ ▝▝ ▗▗▘▖  ▄ ▙ ▘ ▗▗▀  ▘ ▘▝▞▗▗▗▖▝▐▝▚▗▘▝ ▐ ▖▌    ▝▖▘▘▌▀▝   ▘  ▀▞▚▝▗ ▝▗▀▗ ▖▚ ▝ ▖▗▖▘▝▌▐
█▙█▄▟█▄▄▟█▄▟▄▄▄▟█▄▟▟▄▙▙▄▄▄▙▟▙▄▄▄▟▙▄▙▙▟▟▄▄▟██▄▄▙▄▄█▄▄▄▄▄▄▟▄▄▙▟▙▄▄▙▄▄▙▄▙▟▙▄▄▙▄▄▄▄▄▟▟▙▄▄▄▄▙▙█▄▄▄▄▟▟▙▟█▄
```

```
200 - 50 - 19
▌▀▜▀▀▛▜█▛▀▜▜▛▜▀▜▀▀▜▛█▛█▀▛▀▛▀▛▜▀▀▀▀▛▀▀▛▀▀█▜█▀▀▀▜▀▛█▀▜▀▜▛▀▀▀▛▀▀▛▜▛▀█▀▛▛▜▛▜▀▀▀▀▀▀▀▛▀▀▀█▛▀▀▜▀▜▀▛▜▀▀▀▀▀▛▜
▌▛▗▌▝▗▚▜  ▖▝▘▚▖▘  ▌   ▖ ▐ ▟ ▞ ▞▌▛▚ ▚ ▜ ▖▗▛▗  ▝▖▝▗▐▝ ▝  ▄▀▘▖▖▗ ▄ ▖ ▝▞▖▝ ▝▗  ▝▞▘▖▚ ▘▖▖▖▝ ▌▘ ▝▝ ▝▖▗▐ ▐▐
▌  ▐ ▗▀▝ ▗▀▚▝▝▚█▗ ▘▖   ▞▝▗▝▞ ▖▀▟▖  ▘▀▀▌▌▞▖▝▖▗▚▀▐ ▄▐▖▝  ▗ ▖ ▀  ▗▞   ▗▖ ▐ ▝▀▜▞▘▖  ▄ ▞ ▗▜ ▝▘▖▝  ▌▄▝▝ ▄▜
▌▗ ▖▀ ▝  ▖▟ ▖▞▝ ▖▝   ▀▚▚  ▖  ▘▄▗▖▄▘▗ ▗▖▘  ▖▗ ▖ ▄▖  ▗▗▞▝▚▌▗▘▛ ▞     ▗▌▌▗    ▘▘ ▝▞▗▌▝  ▜ ▌▘▚ ▖ ▘▞▐ ▘ ▟
▛ ▘▀▖▗ ▘ ▗    ▌▌▌  ▞ ▝▄▘▖▖▟ ▝ ▄      ▝▗▘▖▝ ▖ ▄ ▖▚▚▘▝▘ ▘▝ ▝▖▖▝▖▜▘▝▞▝▗▐▘▗ ▘▚▘ ▛▖▝  ▐ ▌▄ ▀▝▌ ▐▀▖▜▖  ▗▖▐
▛▗ ▝▝▚▖▖▖▖▖  ▖▖▀ ▞▖ ▘▟▞▝ ▘ ▖ ▖▚▘▗  ▘▗█ ▛▗  ▗▘▖  ▝▘▝ ▌▞ ▞▀ ▌▖▖▖ ▝  ▄▜▞▘▚▖▘▖▜▝▝▌▜▟ ▞▗ ▜▘ ▞  ▝ ▘▘  ▝▐ ▐
▙▖▞▖▗ ▗▗ ▘▄ ▌▘▚▘ ▗▝▗▐  ▐▘▜▄ ▐ ▚▝▌▖▄▌▘▀▘ ▝▖ ▝ ▗▝▖▝▟▙▄▝▗▗ ▞ ▝ ▖   ▜▜▞   ▘▘▌▘▝  ▗ ▌▗▖ ▌▚▗▀▚▝▘▀▄▝ ▚ ▚▘▞▟
▌ ▘ ▗▐▘ ▄▚▖▀▗▘▐  ▘▌    ▝▘▘▝▝ ▘▖▐▘▀▚  ▐▝▝▝▝  ▗▘▗▝▀ ▘ ▖▗▖▗▘▙▘▝  ▌▌  ▖ ▘ ▐▜▖▘▝▖▝▗▌▐ ▘▀▝▗▖ ▘▖▖▌   ▗▀ ▖▘▐
▙▞▖▗▗▟▗▗▚▘ ▘▜ ▘  ▚▟▝▝ ▖▀▝▖ ▖▐ ▘▖▝▙▝ ▝▘▘ ▝  ▝▖▐▐ ▚ ▝ ▝▀▄▗▗▀▘  ▝▚▐▌  ▖▗   ▞ ▚▘▗ ▐▀ ▐▘▝ ▐ ▌▘▞▗▗▝▀▗▘▗▖▐█
▌▗   ▝ ▚▗ ▝▚ ▝ ▝▌ ▖▗ ▝ ▀▖ ▌ ▀▄▝▄▀▐ ▖▗▝ ▝▄  ▚▘▌▖▀▖ ▝▚▘ ▀▄▄▘▜▗▗▘▖▗▌▞▗ ▗ ▘  ▘ ▖  ▀▝▘▙▐▖▐  ▟▝ ▞▗▘▘ ▖▘▐▟▐
▌▌▘▌▌▀▖▝▘▝▘▐    ▗▘▞▀▘ ▘▟▗▘▐  ▀  ▗▝▗ ▘▜▝▟  ▀▖▝ ▖ ▝▞▜▗▌▘   ▗▘ ▘▘ ▝▌▖▘▗▘▝      ▘▗▝ ▘▝▖▖▐▘ ▝▝ ▌ ▞▗▌ ▚▝▌▐
▙▗▗▝▖▜ ▝▖ ▐  ▘▗   ▀ ▌▘▚  ▗▘▖▗▚▚▄▀▞▘▚▗▖▝▘ ▄▖ ▘▝▞▄▖▙ ▘▛ ▗▗ ▘▖ ▗ ▟▗  ▗▖ ▞▖     ▄▖▝▘▝ ▜▝ ▘  ▘▝▌ ▘▀▚▘▗ ▟▜
█ ▖▝ ▘ ▚▞ ▘▖▘▝ ▝▗  ▖▞▟▖▚ ▗▖▘▐ ▖▜▘▝▘▝ ▘ ▚▗ ▘▘▜▖▗▀▝▗▌▗▖ ▖▝ ▀▀▞ ▗▌ ▖▞▄▘▟▚ ▗ █▗▘▝      ▀▖ ▚▗▘ ▖▄▝ ▖  ▗▝▐
▌▘ ▖▝▚▄ ▘▗▚▝ ▝ ▝▗▘▘▘▗   ▄▜▗▝▗▝ ▘ ▗▗  ▖▝▙▗▝▘▘ ▟▟  ▀▖▐  ▐▌▜▘ ▘▘  ▖ ▖█▘▙▟ ▀▜ ▘▄ ▜ ▗ ▚▝ ▟▖▀ ▐   ▞▌▗ ▝▞ ▜
▌ ▖    ▖▝▀  ▝▗  ▝▞ ▚▄▀  ▝▖▗▝▖▖  ▝   ▘ ▗▖ ▚ ▚▄  ▘ ▀▘▜▞▗▗▗▐▌  ▖  ▐▝▗▄  ▝▙▜ ▘▘ ▐▗▝▞▘ ▌▘▘ ▖▝ ▘▟▀▗ ▖▗▄▘ ▟
▛ ▞▘▖  ▜▞  ▘▖ ▗ ▝  ▖▗▗▛▖▝ ▘  ▘▞▜  ▗  ▞   ▖▞ ▙▚▌▝ ▝▗ ▞▘▙ ▘▟ ▖▄▗▗   ▝  ▄▘▌ ▝▘▐▝ ▗▘▗  ▝▘▝ ▘▗▖▜▘▌▝▖▚   ▜
▌▘▘▘▐ ▚▚  ▄▄▘  ▝▖▗▘▖ ▝▐▄  ▖▚▖▐  ▖▖ ▞   ▝  ▝▖▖█ ▗▝ ▞ ▛  ▌▝▝▚▝ ▄  ▖   ▝▖ ▗▐▀  ▝▖  ▖▝▌▗▖▗▝▝▗   ▚ ▝▝  ▘▟
▙▝▘▖▖ ▝▄▖▗ ▗▝▝▘ ▘▘ ▝▝   ▘▐▜▖▗▄ ▝ ▘▌▗▀ ▛▝▟▗ ▘ ▝  ▟▝▌▖▘▝▗ ▘ ▝  ▖▀▛▀▗▗ ▄▙   ▝ ▝▛ ▞▝ ▞▗▙  ▘▐▀ ▐▖▝ ▐▘▗▗▝▜
▌  ▝▖▗ ▚▐ ▗▝ ▄▗▘ ▘▝   ▀▘▄▝ ▚ ▐ ▞▀▌ ▀▘▗▟ ▗▀  ▚  ▖▜▗▝▗ ▘ ▐▘    ▗▖▗  ▐▗▞▗▚▌▖ ▐ ▞▘ ▌   ▚   ▖  ▚▘  ▗ ▚ ▗▐
▌▌▐▄ ▄ ▗▘▀ ▘▘ ▗▄ ▝▗▗▘ ▘▌  ▞▐▖   ▚▐▗ ▝▌    ▐▚▘ ▝   ▖ ▝   ▛▝▗▖▗ ▞▗▚▌▖ ▘▝▘▘▘▖  ▄ ▙▝▀▚▄▀▞▚▘  ▄▖▚▗▝▖    █
▛ ▟▌ ▖ ▝▖ ▐▜   ▄▗▐▗  ▝ ▖▗▚▛ ▝ ▛▘▞▌▗▗▜▖▗▗ ▝ ▜▘ ▀ ▘▌ ▐▝▝▗▘▘▘▀▖▟▟▗▝  ▌  ▝▄▖▝▖▘▘▖▚▄▘▖▗▀▗▝ ▄  ▖▘▀▗▝▗ ▗ ▌▐
▛▝▗   ▖▖▝▚▀▝ ▖ ▚▗▖▗   ▐▙▟▘▀▞▟▗ ▚▖  ▗  ▜▟▘▖▌  ▘▗ ▖▖▘▖ ▟▗ ▝▝▗▝▗ ▞▝▘▝ ▘▝  ▘  ▜ ▘  ▘▟▚▝▘▀▞▝ ▘▀▖▘▄▚ ▗  ▞▐
▌▖▘▝▝▗  ▄▖▝▘▗▘█▚ ▗▘▙ ▌ ▚▘▖▝ ▝   ▞▛▐ ▜▗▞ ▞▗▚▝ ▗▟▌ ▀▚ ▘▗▐▜▗ ▖ ▖  ▀▖▗▖▗█▛▝ ▖ ▗▖▞   ▜  ▝▞▛ ▖▝ ▖  ▗ ▄  ▞▐
▛▖▐ ▙▀ ▞▞▖▝  ▄ ▝  ▘▌▀▝▝▖  ▗▐▖  ▖▖▘▚▀ ▟▗▜▘▗▀▜▌▘▐ ▚▝▘▖▝▖ ▖▝▖▖ ▘  ▚ ▀▜▐▝▘▀▖ ▘▗▖ ▖▙▗▘ ▚  ▜ ▖▘ ▖▝▘▛▗ ▘▟▝█
▙▟▄▄▄▟▄▄▄▄▟▙▄█▙█▄▄▄▄▟▄▄▟▟▄▙▟▙▄▄▄▄▟▄▄▟▄▄▄▙▄▙▄▙▄▙▄▄▟▙▄▄▄▄▟▙▙█▄▟▟▟▄▙▄▄▄▄▄▄▟▙▄▟▙▄▙▄▄▟▟▄▄▄▄▄▟▄▄▟▟▄▄▙▄▟▙▄▄
```

File: W01/test/gad/maze/TestExecutor.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+public class TestExecutor {
+    public static String PATH = "test/gad/maze/test-cases.txt";
```
Comment (felixhauptmann) [inline W01/test/gad/maze/TestExecutor.java:21]:
```suggestion
    public static String PATH = "test/gad/maze/cases.txt";
```

**Review by felixhauptmann — CHANGES_REQUESTED** [review]
Other than that looks fine. I'm passing all tests.

**Review by nilsreichardt — COMMENTED** [review]
Are we going to use upper case folder names for the tasks (I know that Johannes had set up this folder)?

**Comment by Anatoly03** [comment]
@nilsreichardt I believe we can change this in this PR? Or should that be done in another?

**Comment by Anatoly03** [comment]
The last two commits rename the folders to lowercase (see [stackoverflow](https://stackoverflow.com/a/30654833/16002144) why two commits are needed), @nilsreichardt

**Comment by nilsreichardt** [comment]
@JohannesStoehr I think we can this PR because @Anatoly03 fixed @felixhauptmann comment 👍

**Pull request merged**